packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -3243,6 +3337,42 @@ describe("Teal Opcodes", function () {
         RUNTIME_ERRORS.TEAL.INDEX_OUT_OF_BOUND
       );
     });
+
+    it("tealv4: should push correct value accepting offset to foreignAssets", () => {
+      interpreter.tealVersion = 4;
```

Comment:
could you please add tests where tealVersion = 3 or less and value accepted is correct (according to v3)?

Reply:
Previous test follow version <= 3, so that case is already tested. Added new tests for v4.